resurfemg/helper_functions.py

```diff
@@ -1528,3 +1528,47 @@ def find_peaks_in_ecg_signal(ecg_signal, lower_border_percent=50):
         prominence=(max_peak*lower_border_percent/100, max_peak)
     )
     return set_ecg_peaks
+
+
+def variability_maker(array, segment_size, method='variance'):
+    """
+    Calculate variability of segments of an array according to a specific
+    method, then interpolate the values back to the original legnth of array
+
+
+    :param array: the input array
+    :type array: ~numpy.ndarray
+
+    :param segment_size: length over which variabilty calculated
+    :type segment_size: int
+
+    :param method: method for calculation i.e. variance or standard deviation
+    :type method: str
+
+    :returns: variability_values array showing variability over segments
+    :rtype: ~numpy.ndarray
+
+    """
+    variability_values = []
+    if method == 'variance':
+        for i in range(1, len(array) - segment_size):
+            var_emg = np.var(array[i:i + segment_size])
+            variability_values.append(var_emg)
+
+        values_out = np.array(variability_values)
+
+    elif method == 'std':
+        # Calculate the standard deviation of each segment
+        segments = [
+            array[i:i+segment_size] for i in range(0, len(array), segment_size)
+        ]
+        for segment in segments:
+            variability_value = np.std(segment)
+            variability_values.append(variability_value)
+        values_out = np.array(variability_values)
+
+    else:
+        print("You did not choose an exisitng method")
+
+    variability_array = scipy.signal.resample(values_out, len(array))
```

Comment:
I thought about zero padding, but I was afriad we will create an artifact that creates an artificial end to respiratory efforts...needs discussion

Reply:
I think I prefer the 0 padding, as this is only at the end of the signal, and the last breath or effort will usually be skipped anyway so that will not be a problem as artificial end. But the resampling changes the array more over the whole length, right @drcandacemakedamoore ?

Reply:
That is right, the resampling changes the array over all of it. Now the function has a zero-pad option and defaults to an 'average-pad' meaning instead of zeros an average of the last values.